examples/rapids/mlflow_project/src/rf_test/train.py

```diff
@@ -0,0 +1,122 @@
+
+
+def _train(params, fpath, hyperopt=False):
+    """
+    :param params: hyperparameters. Its structure is consistent with how search space is defined. See below.
+    :param fpath: Path or URL for the training data used with the model.
+    :param hyperopt: Use hyperopt for hyperparameter search during training.
+    :return: dict with fields 'loss' (scalar loss) and 'status' (success/failure status of run)
+    """
+    max_depth, max_features, n_estimators = params
+    max_depth, max_features, n_estimators = (int(max_depth), float(max_features), int(n_estimators))
+
+    X_train, X_test, y_train, y_test = load_data(fpath)
+
+    mod = RandomForestClassifier(max_depth=max_depth,
+                                 max_features=max_features,
+                                 n_estimators=n_estimators)
+
+    mod.fit(X_train, y_train)
+    preds = mod.predict(X_test)
+    acc = accuracy_score(y_test, preds)
+
+    mlparams = {"max_depth": str(max_depth),
+                "max_features": str(max_features),
+                "n_estimators": str(n_estimators)}
+    mlflow.log_params(mlparams)
+
+    mlflow.log_metric("accuracy", acc)
+
+    mlflow.sklearn.log_model(mod, "saved_models")
+
+    if (not hyperopt):
+        return mod
+
+    return {'loss': acc, 'status': STATUS_OK}
+
+
+def train(params, fpath, hyperopt=False):
+    """
+    Proxy function used to call _train
+    :param params: hyperparameters. Its structure is consistent with how search space is defined. See below.
+    :param fpath: Path or URL for the training data used with the model.
+    :param hyperopt: Use hyperopt for hyperparameter search during training.
+    :return: dict with fields 'loss' (scalar loss) and 'status' (success/failure status of run)
+    """
+    with mlflow.start_run(nested=True):
+        return _train(params, fpath, hyperopt)
+
+
+if (__name__ == "__main__"):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--algo', default='tpe', choices=['tpe'], type=str)
+    parser.add_argument('--conda-env', required=True, type=str)
+    parser.add_argument('--fpath', required=True, type=str)
+    args = parser.parse_args()
+
+    search_space = [
+        hp.uniform('max_depth', 5, 20),
+        hp.uniform('max_features', 0.1, 1.0),
+        hp.uniform('n_estimators', 150, 1000)
+    ]
+
+    trials = Trials()
+    algorithm = tpe.suggest if args.algo == 'tpe' else None
+    fn = partial(train, fpath=args.fpath, hyperopt=True)
+    experid = 0
+
+    with mlflow.start_run():
+        mlflow.set_tag("mlflow.runName", "RAPIDS-Hyperopt")
+        argmin = fmin(fn=fn,
+                      space=search_space,
+                      algo=algorithm,
+                      max_evals=2,
+                      trials=trials)
+
+        print("===========")
+        fn = partial(train, fpath=args.fpath, hyperopt=False)
+        final_model = fn(tuple(argmin.values()))
+
+        conda_data = ""
+        if (args.conda_env.startswith("http")):
```

Comment:
Interesting, is the ultimate goal here to allow users to log the current conda environment as the model's conda environment? I wonder if it'd work to simply run `conda env export` in a subprocess to get the current conda env in YAML form (we could prob also make it easier to achieve this via MLflow APIs)

Reply:
This is somewhat vestigial in this context, to allow for external environment files to be specified that can be packaged with the saved model.